server/service/src/document/schemas/patient.json

```diff
@@ -174,14 +170,26 @@
         },
         "socioEconomics": {
           "$ref": "#/definitions/SocioEconomics"
+        },
+        "testingArray": {
+          "description": "An array without referenced definitions",
+          "type": "array",
+          "items": {
+            "additionalProperties": false,
+            "properties": {
+              "field1": {
+                "description": "First item",
+                "type": "string"
+              },
+              "field2": {
+                "description": "Second item",
+                "type": "string"
+              }
+            }
+          }
```

Comment:
Why does Rust even need to know what the structure of a doc schema is -- it could be anything and change at any time right?

It doesn't need to be part of any "real" patient data or anything, it's just there so I can test the range of possibilities in the JSON Forms component. In this case, it refers to a field that is missing in the example patient data documents, so we can ensure that the form builder displays the fields correctly anyway (not obvious, since the Array component iterates over existing data element, so this just ensures we've remembered to create an empty value for this field in the data object if its not defined)

Reply:
Ah, I see from your later comment that you're using this schema to create the Patient data type. I don't know that we should rely on a document schema for this. A real-word patient document might contain a whole bunch of fields that are not part of the "core" patient data type, wouldn't it?

Reply:
There is a base patient schema that all derived schema should be compatible to. I use this base patient schema to derive the rust type, e.g. to parse the json data as a patient object and then fill the names table from it.

When updating a patient I would actually have to check that the data fulfills the base patient schema and the actual patient schema for the data.

So you could actually put your custom patient schema into the schema table during initialization and that should work as well (don't touch the schema in service).

Reply:
> So you could actually put your custom patient schema into the schema table during initialization and that should work as well (don't touch the schema in service).

Yeah, that's probably better for now. I'll change it.

We'll sort out a "definitive" json schema in the next few days -- the current one is intended to test a few edge cases.